netlink-packet-route/src/rtnl/tc/nlas/class/mod.rs

```diff
@@ -0,0 +1,42 @@
+// SPDX-License-Identifier: MIT
+
+use crate::{
+    nlas::{self, NlaBuffer},
+    traits::{ParseableParametrized},
+    DecodeError,
+};
+
+pub const HTB_CLASS_KIND: &str = "htb";
+
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub enum Class {
+    Other(Vec<u8>),
+}
+
+impl nlas::Nla for Class {
+    fn value_len(&self) -> usize {
+        match self {
+            Self::Other(o) => o.len(),
+        }
+    }
+
+    fn emit_value(&self, buffer: &mut [u8]) {
+        match self {
+            Self::Other(o) => buffer.copy_from_slice(o.as_slice()),
+        }
+    }
+
+    fn kind(&self) -> u16 {
+        unreachable!("the parent nla will return TCA_OPTIONS")
```

Comment:
I'd rather not panic in a publicly available method.

If I understand correctly, `Class` is not an NLA, but rather the payload of an `TCA_OPTIONS` NLA. In that case, I think it shouldn't implement the `Nla` trait (and same goes for `Filter` and `Qdisc`. Or am I mis-understanding something?

Reply:
The `TcOpt` is used for different message type distribution. Just use `parse_with_param` and pass in the `kind` parameter to distinguish.

Reply:
I like your new implementation better, it minimizes the changes in `netlink-packet-route`. Thanks for doing this!